amplitude.js

```diff
@@ -270,15 +253,130 @@ Amplitude.prototype.runQueuedFunctions = function () {
   this._q = []; // clear function queue after running
 };
 
-Amplitude.prototype._loadSavedUnsentEvents = function(unsentKey, queue) {
-  var savedUnsentEventsString = localStorage.getItem(unsentKey);
+/**
+ * Set an item in local storage. Decorate item key with the api key
+ * to avoid collisions.
+ *
+ * @property item: name of the item to store
+ * @property value: the value to store
+ */
+Amplitude.prototype.setLocalStorage = function(item, value) {
+  var key = LocalStorageKeys[item] + '_' + this.options.apiKey.slice(0, 6);
+  localStorage.setItem(key, value);
+};
+
+/**
+ * Fetch an item from local storage. Decorate item key with api key
+ * to avoid collisions.
+ *
+ * @property item: name of the item to fetch
+ * @property def: default value to return if the item does not exist
+ */
+Amplitude.prototype.getLocalStorage = function(item, def) {
```

Comment:
default is a reserved word in JS. I'll change it to something like default_value